spring-boot-examples/workflows/README.md

```diff
@@ -0,0 +1,403 @@
+<!-- STEP
+name: Run Demo Workflow Application
+match_order: none
+output_match_mode: substring
+expected_stdout_lines:
+- 'Started WorkflowPatternsApplication'
+background: true
+expected_return_code: 143
+sleep: 30
+timeout_seconds: 45
+-->
+<!-- Timeout for above service must be more than sleep + timeout for the client-->
+
+```sh
+../../mvnw spring-boot:test-run
+```
+
+<!-- END_STEP -->
+
+Once the application is running you can trigger the different patterns by sending the following requests: 
+
+### Chaining Activities Workflow example
+
+The `io.dapr.springboot.examples.wfp.chain.ChainWorkflow` executes three chained activities. For this example the 
+`ToUpperCaseActivity.java` is used to transform to upper case three strings from an array.
+
+```mermaid
+graph LR
+   SW((Start
+   Workflow))
+   A1[Activity1]
+   A2[Activity2]
+   A3[Activity3]
+   EW((End
+   Workflow))
+   SW --> A1
+   A1 --> A2
+   A2 --> A3
+   A3 --> EW
+```
+
+<!-- STEP
+name: Start Chain Activities Workflow
+match_order: none
+output_match_mode: substring
+expected_stdout_lines:
+- 'TOKYO, LONDON, SEATTLE'
+background: true
+sleep: 1
+timeout_seconds: 2
+-->
+<!-- Timeout for above service must be more than sleep + timeout for the client-->
+
+To start the workflow with the three chained activities you can run: 
+
+```sh
+curl -X POST localhost:8080/wfp/chain -H 'Content-Type: application/json' 
+```
+
+<!-- END_STEP -->
+
+
+As result from executing the request you should see: 
+
+```bash
+TOKYO, LONDON, SEATTLE
+```
+
+In the application output you should see the workflow activities being executed. 
+
+```bash
+io.dapr.workflows.WorkflowContext        : Starting Workflow: io.dapr.springboot.examples.wfp.chain.ChainWorkflow
+i.d.s.e.w.WorkflowPatternsRestController : Workflow instance 7625b4af-8c04-408a-93dc-bad753466e43 started
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Starting Activity: io.dapr.springboot.examples.wfp.chain.ToUpperCaseActivity
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Message Received from input: Tokyo
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Sending message to output: TOKYO
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Starting Activity: io.dapr.springboot.examples.wfp.chain.ToUpperCaseActivity
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Message Received from input: London
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Sending message to output: LONDON
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Starting Activity: io.dapr.springboot.examples.wfp.chain.ToUpperCaseActivity
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Message Received from input: Seattle
+i.d.s.e.wfp.chain.ToUpperCaseActivity    : Sending message to output: SEATTLE
+io.dapr.workflows.WorkflowContext        : Workflow finished with result: TOKYO, LONDON, SEATTLE
+```
+
+### Parent / Child Workflows example
+
+In this example we start a Parent workflow that calls a child workflow that execute one activity that reverse the . 
+
+The Parent workflow looks like this: 
+
+```mermaid
```

Comment:
Did you mean to put `mermaid` here?

Reply:
@cicoyle yes.. can you please elaborate on the question?

Reply:
oh, I see its to have a graph. didn't render the markdown locally and was confused to see `mermaid` there